ansible/roles/gitlab_runner/defaults/main.yml

```diff
@@ -171,24 +171,38 @@ gitlab_runner__api_url: 'https://{{ gitlab_runner__api_fqdn }}/'
                                                                    # ]]]
 # .. envvar:: gitlab_runner__api_token [[[
 #
-# The personal GitLab API access token used for API operations, for example
-# removal of existing Runners. This is not a Runner registration token. You can
-# generate an access token in GitLab "User Settings", "Access Tokens"
+# The personal GitLab API access token used to register or remove the runners in GitLab.
+# You can generate an access token in GitLab "User Settings", "Access Tokens"
 # configuration page.
 gitlab_runner__api_token: ''
 
                                                                    # ]]]
-# .. envvar:: gitlab_runner__executor [[[
+# .. envvar:: gitlab_runner__runner_type [[[
 #
-# Default "executor" used by the runners.
-gitlab_runner__executor: 'shell'
+# The scope of the runner.
+# It can be one of the following values: ``instance_type``,
+# ``group_type`` or ``project_type``
+gitlab_runner__runner_type: 'instance_type'
+
+                                                                   # ]]]
+# .. envvar:: gitlab_runner__group_id [[[
+#
+# The ID of the group that the runner is created in.
+# Required if ``gitlab_runner__runner_type`` is ``group_type``.
+gitlab_runner__group_id: ~
 
                                                                    # ]]]
-# .. envvar:: gitlab_runner__token [[[
+# .. envvar:: gitlab_runner__project_id [[[
 #
-# The GitLab CI registration token used to register the runners in GitLab.
-# See :ref:`gitlab_runner__token` for more details.
-gitlab_runner__token: '{{ lookup("env", "GITLAB_RUNNER_TOKEN") }}'
```

Comment:
It would be useful to have this kind of lookup in `gitlab_runner__api_token` variable. Since runners are per-user, the token can be set in the `.env` file of the DebOps project directory, and then at runtime the variable will be available in the environment.

Reply:
Fixed